### ISSUE TITLE
Choose a development stack

### ISSUE THREAD

**Issue by Decafoats** [Decafoats/Achilles#1]
The first issue of the Repo (Scrapbook moment 📷)!

As per our [milestone #1](https://github.com/Decafoats/Achilles/milestone/1), we must decide on a development stack for this capstone project.

There are many to choose from regarding app development, Flutter, JS, Java, Kotlin, etc. @AhrasAli and I will do extensive research and decide on the best stack to use moving forward.

**Comment by Decafoats** [comment]
After doing some research, @AhrasAli and I narrowed it down to two languages to choose from for our app development stack. 

**Flutter vs Java.**

# Flutter

## Pros

- Open-source
- Backed by Google and runs quickly on the Google Cloud Platform.
- Dart is approximately two times faster than JavaScript.
- Dart is type-safe and compiled with both AOT and JIT compilers.
- Dart is very scalable across projects.
- Dart is used extensively for the Flutter mobile UI framework.

## Cons 

- Dart is relatively new to programmers and rarely used in the market.
- Dart has minimal resources online, and finding solutions to problems is complicated.
- Dart has a single object paradigm (classes).
- In Dart, you can’t rename the function without writing another assignment statement.

# Javascript

## Pros

- JavaScript can be used for both web and mobile apps.
- It can be used for both frontend and backend so that JavaScript can run on every device.
- JavaScript has a vast community and remarkable frameworks available online.
- JavaScript is friendly to other languages, so that many other apps can use JavaScript.
- JavaScript is fast, flexible, and lightweight.

## Cons

- JavaScript has some libraries that are not of good quality.
- Being a dynamic language, programmers can make big mistakes easily.
- There are constant changes as new frameworks land after a regular interval.
- Client-side security leaves a lot to be desired.
- One error can crash an entire website.
- Multi-threading support is not available.

In the end, @AhrasAli and I decided to go with Javascript. The reason is that from my previous experience with both flutter and Javascript, I have found that Javascript is easier to learn and has excellent documentation and online resources. Flutter, on the other hand, might be fancier, but it was a pain when I got stuck trying to implement certain features and ended up wasting hours on Googling solutions. As we are limited in time for this capstone project, learning a new language from scratch with poor documentation seems like a digging hole for ourselves in the long run.

TL;DR - JAVASCRIPT (React Native) IS THE WINNER! 🎉
